### PR TITLE
Implement Flatten

### PR DESCRIPTION
**Pull request by yukinarit**
```python
@deserialize
 @serialize
 @dataclass
 class Bar:
     c: float
     d: bool

 @deserialize
 @serialize
 @dataclass
 class Foo:
     a: int
     b: str
     bar: Bar = field(metadata={'serde_flatten': True})

 def main():
     f = Foo(a=10, b='foo', bar=Bar(c=100.0, d=True))
     print(f"Into Json: {to_json(f)}")

     s = '{"a": 10, "b": "foo", "c": 100.0, "d": true}'
     print(f"From Json: {from_json(Foo, s)}")
```

### CODE REVIEW

File: serde/de.py [hunk 1]
```diff
@@ -487,7 +487,20 @@ def custom_field_deserializer(self, arg: DeField) -> str:
         return f"{arg.deserializer.name}({arg.data})"
 
     def dataclass(self, arg: DeField) -> str:
-        return f"{arg.type.__name__}.{SERDE_SCOPE}.funcs['{self.func}']({arg.data}, reuse_instances=reuse_instances)"
+        if not arg.flatten:
+            # e.g. "data['field']" will be used as variable name.
+            var = arg.data
+        else:
+            # Because the field is flattened.,
```
Comment (yukinarit) [inline serde/de.py:494]: Remove comma

**Comment by yukinarit** [comment]
Hi @ydylla  

I have implemented to flatten feature. PTAL!

**Review by ydylla — APPROVED** [review]
Looks good.

**Comment by yukinarit** [comment]
@ydylla thank you!

**Pull request merged**